accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,216 @@
+
+* Alerts and Recommendations via Loki
+    * Loki Querying: Utilize the Loki API to query log data for patterns indicative of issues or anomalies. This requires crafting LogQL queries tailored to the types of issues Uyuni Health-Check-Tool aims to detect.
+    * Alert Generation: Employ the Loki Ruler component to define alerting rules. These rules can be dynamically generated or updated based on the Uyuni Health-Check-Tool configuration, allowing for customizable alert conditions.
+
+* Implementing alerts and recommendations involves:
+    * Defining LogQL Queries: Writing LogQL queries that match the relevant conditions.
+    * Configuring Alert Rules: Using the Loki Ruler to define alert rules based on LogQL queries. These rules specify the conditions under which alerts should be triggered and the severity levels.
+    * Setting Up Notification Channels: Configuring the notification channels to receive alerts.
+    * Recommendation Logic: For recommendations, the alerting mechanism can be extended with additional logic to suggest actions. 
+  
+
+## Metrics
+
+* Server status and resource utilization
+
+* Database health and integrity
+
+* Status of relevant Uyuni services
+
+  
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+  
+
+* Resource Usage: Running additional services for log aggregation, analysis, and enhanced alerting could increase the resource demands on the server, including CPU, memory, and storage. This might necessitate upgrades to existing hardware or reevaluation of resource allocation in cloud environments to ensure optimal performance.
+
+* Learning Curve: System administrators and developers may need to familiarize themselves with Loki/Promtail's log querying language (LogQL) and Grafana's alerting mechanisms. The learning curve associated with these tools could slow down initial adoption and efficiency gains.
+
+* Security Considerations: With the introduction of new components for logging and alerting, there might be additional security considerations to address, including access controls, data encryption, and secure communication channels between services.
+
+* Alert Noise: Fine-tuning alert rules to balance sensitivity and specificity is critical. Without careful configuration, there's a risk of alert fatigue due to a high volume of non-critical alerts, potentially causing important alerts to be overlooked.
+
+  
+
+# Alternatives
+
+[alternatives]: #alternatives
+
+  
+
+- What other designs/options have been considered?
+
+Running the Uyuni Health-Check-Tool along with tye Uyuni Server container in the same pod. This could lead to a resource problem on the node running the pod.
+
+- What is the impact of not doing this?
+
+
+
+  
+
+# Unresolved questions
+
+[unresolved]: #unresolved-questions
+
+
+- What are the unknowns?
+
+Has Saline to be part of the current Uyuni Monitoring stack?
+
+How is the exporter going to access to uyuni-server container? For example, to query the database.
```

Comment:
The current design assumes this will work. If it does not, a lot will need to be reconsidered.

Reply:
Correct. I've described this situation a bit more in the RFC text.